File: code/backend/src/main/java/project/backend/dto/RecipeFilterDto.java
package project.backend.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import project.backend.enums.RecipeFilterCriterias;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class RecipeFilterDto {

    private String name;

    private String category;

    private String mealType;

    private int maxPreparationTime;

    private int minDifficulty;

    private int maxDifficulty;

    private String[] tags;

    private IngredientDto[] ingredients;

    private RecipeFilterCriterias filterCriteria;
}

File: code/frontend/src/app/dtos/recipe-filter-dto.ts
import { IngredientDto } from "./ingredient-dto";

export interface RecipeFilterDto {
  name: string;
  category: string;
  mealType: string;
  maxPreparationTime: number;
  minDifficulty: number;
  maxDifficulty: number;
  tags: string[];
  ingredients: IngredientDto[];
  filterCriteria: string;
}
